pkg/cmd/server/api/types.go

```diff
@@ -533,7 +533,12 @@ type KubernetesMasterConfig struct {
 	APILevels []string
 	// MasterIP is the public IP address of kubernetes stuff.  If empty, the first result from net.InterfaceAddrs will be used.
 	MasterIP string
```

Comment:
before we introduce parallel variables for things we might want to load from envvars, we should decide if we want parallel fields for everything we might want to do that for. Things that come to mind:

1. HA-specific fields, like IPs/hostnames
2. Secret fields, like `clientSecret` for OAuth identity providers, or `bindPassword` for the upcoming LDAP identity provider

I would sort of like to let those fields take direct values, or envvars, or values from a conf file, or some other yet undetermined mechanism (etcd lookup, etc). @deads2k is opposed to letting values contain a syntax like `${ENVVAR}`, `${FILE:/path/to/conf/file:KEY_NAME}`. I sort of like the unification, but I can see pitfalls. Implications (good, bad, and neutral):
* you can only do it for string fields (or yaml unmarshalling won't preserve the value)
* a single field is sufficient for any number of resolution sources
* you have to be careful not to resolve values when reading a config in order to mutate and write it back out (we don't do this yet, but we may someday)
* you have to be careful to resolve values before using/validating the config